common/txmgr/types/config.go

```diff
@@ -0,0 +1,58 @@
+package types
+
+import "time"
+
+type TxmConfig[UNIT any] interface {
+	BroadcasterConfig[UNIT]
+	ConfirmerConfig[UNIT]
+	ResenderConfig
+	ReaperConfig
+
+	SequenceAutoSync() bool
+	UseForwarders() bool
+	MaxQueuedTransactions() uint64
+}
+
+type BroadcasterConfig[UNIT any] interface {
+	TriggerFallbackDBPollInterval() time.Duration
+	MaxInFlightTransactions() uint32
+
+	// from gas.Config
+	IsL2() bool
+	MaxFeePrice() UNIT
+	FeePriceDefault() UNIT
+}
+
+type ConfirmerConfig[UNIT any] interface {
+	RPCDefaultBatchSize() uint32
+	UseForwarders() bool
+	FeeBumpTxDepth() uint16
+	MaxInFlightTransactions() uint32
+	FeeLimitDefault() uint32
+
+	// gas config
+	FeeBumpThreshold() uint64
+	FinalityDepth() uint32
```

Comment:
nit: FinalityDepth is not just gas config. Perhaps remove it from this grouping.

Reply:
true, it is used for more than just gas, but it those methods come from the original `gas.Config` interface hence the comment - but I can clarify that comment a bit